pinc/gradual.inc

```diff
@@ -227,3 +228,144 @@ function maybe_output_new_proofer_project_message($project)
         echo "</div>";
     }
 }
+
+/*
+ * Output a callout and encourage the user to work in the round with the
+ * highest backlog that they can work in.
+ */
+function encourage_highest_round($username, $round_id = null)
+{
+    global $code_url, $Round_for_round_id_, $ELR_round, $ACCESS_CRITERIA;
+
+    // This URL fragment gets used a lot in this file
+    $round_url = "$code_url/tools/proofers/round.php?round_id=";
+
+    // get the backlog stats for all rounds sorted largest first
+    $backlog_stats = get_round_backlog_stats(array_keys($Round_for_round_id_));
+    arsort($backlog_stats);
+
+    // get all the rounds a user can work
+    $rounds_can_work_in = array_keys(get_stages_user_can_work_in($username));
+
+    // loop through the needs, highest to lowest, and point them to the
+    // one where they can make the most impact with their current access
+    foreach (array_keys($backlog_stats) as $round_target) {
+        $round = get_Round_for_round_id($round_target);
+
+        // don't encourage people towards $ELR_round, as that isn't really
+        // helpful for most sites
+        if ($round->id == $ELR_round->id) {
+            continue;
+        }
+
+        // if $round_id is the current one, ignore $round_id because that means
+        // we're on that round's page already
+        if ($round_target == $round_id) {
+            return;
+        }
+
+        // prepare the common wordings between the callouts
+        $please_move_up = sprintf(
+            _("Please consider proofreading a project in $round_target. Having eligible users working in <a href='%s'>%s</a> helps significantly with our backlog."),
+            "$round_url$round_target",
+            $round_target
+        );
+
+        // They can work in the target round, encourage them to do so.
+        if (in_array($round_target, $rounds_can_work_in)) {
+            echo "<div class='callout'>";
+            echo "<div class='calloutheader'>";
+            echo sprintf(_("Consider working on projects in %s!"), $round_target);
+            echo "</div>";
+
+            echo "<p>$please_move_up</p>";
+            echo "</div>";
+            return;
+        }
+
+        // They're already eligible but have not requested access, tell them.
+        $uao = $round->user_access($username);
+        if ($uao->all_minima_satisfied) {
+            echo "<div class='callout'>";
+            echo "<div class='calloutheader'>";
+            echo sprintf(_("You are eligible to work in %s!"), $round_target);
+            echo "</div>";
+
+            echo "<p>";
+            echo _("You are eligible to work on projects in $round_target!") . " ";
+            if ($round->after_satisfying_minima == 'REQ-AUTO') {
+                echo sprintf(
+                    _("Simply go to the <a href='%s'>%s</a> page to be automatically granted access, find a project that interests you, and start proofreading."),
+                    "$round_url$round_target",
```

Comment:
same as above and throughout in html string generation

Reply:
Interesting. Quick first pass, not detailed testing (and I haven't read through the code in detail, which would answer the question, I'm sure) -- are the F1 and F2 callout boxes supposed to be recommending proofreading in P3?

Reply:
Yes, because it's always going to recommend people to the round with the highest backlog (that they can work in), which the sandbox thinks is P3 per the `["P3", "F2", "F1", "P1", "P2"]` hack mentioned in the description. That's also true for PROD.

Reply:
Ah, okay, that makes sense.

Reply:
I guess that applies to my next point of confusion (comment below) -- I may have "access" to F1 now, but P1 still has the largest backlog on TEST.